client/src/connection_cache.rs

```diff
@@ -78,6 +83,18 @@ impl ConnectionCache {
         }
     }
 
+    /// Create a connection cache with a specific quic client endpoint.
+    pub fn new_with_endpoint(connection_pool_size: usize, client_endpoint: Endpoint) -> Self {
+        // The minimum pool size is 1.
+        let connection_pool_size = 1.max(connection_pool_size);
```

Comment:
Could we make this duplicated code less brittle by adding a non-pub method that both `ConnectionCache::new()` and `ConnectionCache::new_with_endpoint()` call, like:

```
pub fn new(connection_pool_size: usize) -> Self {
    Self::_new_with_endpoint(connection_pool_size, None)
}

pub fn new_with_endpoint(connection_pool_size: usize, client_endpoint: Endpoint) -> Self {
    Self::_new_with_endpoint(connection_pool_size, Some(client_endpoint)
}

fn _new_with_endpoint(connection_pool_size: usize, client_endpoint: Option<Endpoint>) -> Self {
    // The minimum pool size is 1.
    let connection_pool_size = 1.max(connection_pool_size);
    Self {
        use_quic: true,
        connection_pool_size,
        client_endpoint,
        ..Self::default()
    }
}
```

Reply:
Done